core/backend/src/CheckpointManager.ts

```diff
@@ -208,19 +208,25 @@ export class V2CheckpointManager {
       if (!container.isConnected)
         container.connect(this.cloudCache);
       container.checkForChanges();
+      const dbStats = container.queryDatabase(dbName);
       if (IModelHost.appWorkspace.settings.getBoolean("Checkpoints/prefetch", false)) {
         const getPrefetchConfig = (name: string, defaultVal: number) => IModelHost.appWorkspace.settings.getNumber(`Checkpoints/prefetch/${name}`, defaultVal);
         const minRequests = getPrefetchConfig("minRequests", 3);
         const maxRequests = getPrefetchConfig("maxRequests", 6);
         const timeout = getPrefetchConfig("timeout", 100);
-        const logPrefetch = async (prefetch: CloudSqlite.CloudPrefetch) => {
-          const stopwatch = new StopWatch(`[${container.containerId}/${dbName}]`, true);
-          Logger.logInfo(loggerCategory, `Starting prefetch of ${stopwatch.description}`, { minRequests, maxRequests, timeout });
-          const done = await prefetch.promise;
-          Logger.logInfo(loggerCategory, `Prefetch of ${stopwatch.description} complete=${done} (${stopwatch.elapsedSeconds} seconds)`, { minRequests, maxRequests, timeout });
-        };
-        // eslint-disable-next-line @typescript-eslint/no-floating-promises
-        logPrefetch(CloudSqlite.startCloudPrefetch(container, dbName, { minRequests, nRequests: maxRequests, timeout }));
+        const maxBlocks = getPrefetchConfig("maxBlocks", 500); // default size of 2GB. Assumes a checkpoint block size of 4MB.
+        if (dbStats?.totalBlocks !== undefined && dbStats?.totalBlocks <= maxBlocks) {
```

Comment:
2nd "?" is not necessary, 2G seems pretty small. No?

Reply:
Oops, removed the 2nd "?". Yeah I would say 2G is probably too small. I wanted to leave a conservative number for the default, and maybe 2G would be the starting point. The number will go up through settings provided by backends (which is controlled by a feature flag).